Add clarification for backends persistent storage (#340)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/con_primary-parameters-of-the-servicetelemetry-object.adoc

        enabled: true
----

[id="backends-configuring-persistent-storage-for-prometheus_{context}"]
[discrete]
=== Configuring persistent storage for Prometheus

> Use the additional parameters defined in `backends.metrics.prometheus.storage.persistent` to configure persistent storage options for Prometheus, such as storage class and volume size.

Reviewer: ```suggestion
Use the additional parameters that are defined in `backends.metrics.prometheus.storage.persistent` to configure persistent storage options for Prometheus, such as storage class and volume size.
```

> Use `storageClass` to define the backend storage class. Service Telemetry Operator uses the default defined storage class for the {OpenShift} cluster if this is not defined.

Reviewer: ```suggestion
Use `storageClass` to define the backend storage class. If you do not set this parameter, Service Telemetry Operator uses the default defined storage class for the {OpenShift} cluster.
```

Author: Same update here as above.

> Use `storageClass` to define the backend storage class. Service Telemetry Operator will use the default defined storage class for the {OpenShift} cluster if this is not defined.

Reviewer: ```suggestion
Use `storageClass` to define the backend storage class. Service Telemetry Operator uses the default defined storage class for the {OpenShift} cluster if this is not defined.
```

> Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it is possible that a volume size larger than requested is used. By default, Service Telemetry Operator requests a volume size of `20G` (20 Gigabytes).

Author: ```suggestion
Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it is possible that a volume size larger than requested is used. By default, Service Telemetry Operator requests a volume size of `20G` (20 gigabytes).
```

> Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it's possible a volume size larger than requested may be used. By default Service Telemetry Operator requests a volume size of `20G` (20 Gigabytes).

Reviewer: ```suggestion
Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it is possible that a volume size larger than requested is used. By default, Service Telemetry Operator requests a volume size of `20G` (20 Gigabytes).
```

> Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it is possible that a volume size larger than requested is used. By default, Service Telemetry Operator requests a volume size of `20G` (20 Gigabytes).

Reviewer: ```suggestion
Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it is possible that a volume size larger than requested is used. By default, Service Telemetry Operator requests a volume size of `20Gi` (20 Gigabytes).
```

        enabled: true
----

[id="backends-configuring-persistent-storage-for-elasticsearch_{context}"]
[discrete]
=== Configuring persistent storage for ElasticSearch

Use the additional parameters defined in `backends.events.elasticsearch.storage.persistent` to configure persistent storage options for ElasticSearch, such as storage class and volume size.

> Use `storageClass` to define the backend storage class. Service Telemetry Operator will use the default defined storage class for the {OpenShift} cluster if this is not defined.

Reviewer: ```suggestion
Use `storageClass` to define the backend storage class. If you do not define this parameter, the Service Telemetry Operator uses the default defined storage class for the {OpenShift} cluster.
```

Author: Fixed my use of "backend" to be "back end" for consistency with the recent reviews.

Author: I also dropped the word "defined" as it didn't add value. Let me know if you like this adjustment. If so, feel free to commit!

> Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it's possible a volume size larger than requested may be used. By default Service Telemetry Operator requests a volume size of `20Gi` (20 Gigabits).

Reviewer: ```suggestion
Use the `pvcStorageRequest` parameter to define the minimum required volume size to satisfy the storage request. If volumes are statically defined, it is possible that a volume size larger than requested is used. By default, Service Telemetry Operator requests a volume size of `20Gi` (20 Gigabits).
```

Reviewer: @leifmadsen Just want to double check with you that Gigabits is correct her.

Author: Thank you for asking! You are indeed correct that it's not Gigabits. It's actually Gibibytes, which is the representation of power of 2 vs power of 10 (per my searching).

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/installation_guide/data-size-appendix

https://searchstorage.techtarget.com/definition/gibibyte-GiB
